adapters/repos/db/vector/hnsw/index_too_many_links_bug_integration_test.go

```diff
@@ -263,6 +260,21 @@ func Test_NoRace_ManySmallCommitlogs(t *testing.T) {
 		}
 	})
 
+	t.Run("verify that the loaded index has no duplicated connections", func(t *testing.T) {
```

Comment:
Shouldn't we wait for the tombstone cleanup to finish before running this part of the test? When testing here without your fix there were no duplicates yet because the tombstone cleanup was still running.